src/components/organisms/add-product-modal/index.tsx

```diff
@@ -16,9 +16,14 @@ const AddProductModal: React.FC<AddProductModalProps> = ({
 }) => {
   const [selectedProducts, setSelectedProducts] = React.useState<any[]>([])
 
-  const handleSelectProduct = () => {
+  const handleSelectProduct = async () => {
     if (selectedProducts.length > 0) {
-      onSubmit(selectedProducts.map((p) => p.id))
+      const allIds = collectionProducts.map((product) => product.id)
+
+      const selectedIds = selectedProducts.map((product) => product.id)
+      const removedIds = allIds.filter((id) => !selectedIds.includes(id))
+
+      await onSubmit(selectedIds, removedIds)
```

Comment:
This is not exactly what we want to do - consider this:
- you have 5 pages worth of products
- you want to add 2 products to the collection; the two products are on the first page
- all the existing products in the collection are on page 4
- you check the two products on page one and click save
- you now only have the 2 products on page 1 in your collection

What we need to do is to keep track of both which products we have checked but also the products which you have unchecked so that we don't do a `filter` when finding the `removedIds` but rather that we have these available in a state variable similar to how `selectedProducts` are controlled